spacy/vectors.pyx

```diff
@@ -357,8 +361,10 @@ cdef class Vectors:
         # Account for numerical error we want to return in range -1, 1
         scores = xp.clip(scores, a_min=-1, a_max=1, out=scores)
         row2key = {row: key for key, row in self.key2row.items()}
+
+        numpy_rows = get_current_ops().to_numpy(best_rows)
```

Comment:
`xp` might not be the same as `get_current_ops`, I think?

Reply:
Nevermind, it shouldn't matter...

Reply:
This could only go wrong if for some reason `vectors.data` is on GPU while the current ops is `NumpyOps` - I don't really see how that could happen? Still, we could call `CupyOps.to_numpy()` directly, which will always work.

Reply:
You can totally have:

```python
spacy.require_gpu()
nlp = spacy.load("model")
spacy.require_cpu()
nlp.do_something()
```

The thinc models should keep working in this context. (The torch ones don't because of the default tensor type.)

It probably doesn't come up too often, if it's just the rows, then maybe `xp.asarray()` would be better? (I don't know how much data would theoretically be copied here?)

Reply:
(Wait, that's nonsense.)

Reply:
`to_numpy` is fine, it's in the base `Ops`.

Reply:
Yes, but the base ops only works on numpy arrays. I'm confused now whether this issue is resolved or not :p

Reply:
Argh, I just keep confusing myself, sorry.